fix application uninstallation

Xento opened the pull request:
Removes the enforcement deadline from the assignments, so that an uninstallation can be triggered.

After that the enforcement deadlines are set to the old value.

This function is very helpful when you want to reinstall an application.
So you don't have to change the collectionmeberships and wait that the client will get the new policies.

Commit:
Removes the enforcement deadline from the assignments, so that an uninstallation can be triggered.
After that the enforcement deadlines are set to the old value.

This function is very helpful when you want to reinstall an application.
So you don't have to change the collectionmeberships and wait that the client will get the new policies.

rzander reviewed (approved):
THX, I'm not sure if the 10s and 30s sleep is required ? Maybe an "evaluate machine policy" will reset the changes without waiting that Long...

Xento commented:
After removing the deadlines I had to wait a littlebit to get populated and than I could fire the uninstallation.
After this you can set the deadlines to default value.
Maybe shorter witing than 10 seconds is possible I didn't tried shorter values.
It is a littlebit ago since I added this, so I don't so I can't remember why I added 30 seconds in the catch case.
